src/Illuminate/Support/Collection.php

```diff
@@ -54,6 +54,21 @@ public function all()
     }
 
     /**
+     * Get the average value of a given key.
+     *
+     * @param  string|null  $key
+     * @return mixed
+     */
+    public function avg($key = null)
+    {
+        if ($this->count() == 0) {
```

Comment:
No this is not what I check
http://stackoverflow.com/questions/6741399/can-i-define-a-variable-in-a-php-if-condition